base/gmp.jl

```diff
@@ -95,12 +95,18 @@ function __init__()
               cglobal(:jl_gc_counted_malloc),
               cglobal(:jl_gc_counted_realloc_with_old_size),
               cglobal(:jl_gc_counted_free_with_size))
-
         ZERO.alloc, ZERO.size, ZERO.d = 0, 0, C_NULL
         ONE.alloc, ONE.size, ONE.d = 1, 1, pointer(_ONE)
     catch ex
         Base.showerror_nostdio(ex, "WARNING: Error during initialization of module GMP")
     end
+    # This only works with a patched version of GMP, ignore otherwise
+    try
+        ccall((:__gmp_set_alloc_overflow_function, :libgmp), Cvoid,
+              (Ptr{Cvoid},),
+              cglobal(:jl_throw_out_of_memory_error))
+    catch
```

Comment:
Can the type of the exception be checked here (in cases someone modifies this code and perhaps introduces an undefined variable error)?

Reply:
Done